pkg/admission/certs.go

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package admission
+
+import (
+	"crypto/tls"
+
+	"k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+// CertsConfig is the configuration for certificate generation and consumption.
+type CertsConfig struct {
+	// Place to host the certs. It can be empty if the certs haven't been generated.
+	Certs *Certs
+	// Configuration for generating the certs.
+	GenConfig *GenCertConfig
+	// Configuration for consuming the certs.
+	ConsumeConfig *ConsumeCertConfig
+}
+
+// Certs hosts the Server Certificate and the CA certificate.
+type Certs struct {
+	CACert     *tls.Certificate
+	ServerCert *tls.Certificate
+}
+
+// GenCertConfig is the configuration for generating certificates.
+type GenCertConfig struct {
+	DNSName      []string
+	Organization string
+}
+
+// ConsumeCertConfig is the configuration of how the certificate can be consumed.
+type ConsumeCertConfig struct {
+	SecretName      *string
+	SecretNamespace *string
+
+	ServerCertPath *string
+	ServerKeyPath  *string
+
+	TLSConfig *tls.Config
+}
+
+type TLSConfigGetter interface {
+	// Get returns the TLS config (private key and cert), the CA cert and an potential error.
+	Get() (*ConsumeCertConfig, error)
```

Comment:
This part confuses me. The interface is called `TLSConfigGetter`, but Get returns me a `ConsumeCertConfig`? Can you change the name to be consistent?